network/gopubsub/pubsub.go

```diff
@@ -0,0 +1,126 @@
+package gopubsub
+
+import (
+	"fmt"
+	"strconv"
+	"sync"
+)
+
+//PubSubChannel represents a pubsub system where subscriber can .Subscribe() and publishers can .Publish() or .Close().
+type PubSubChannel struct {
+	subscriptions []*Subscription
+	subsMutex     sync.RWMutex
+	nextId        int
+}
+
+//Subscription is created in PubSubChannel using pubSub.Subscribe(). Subscribers can receive using .ReceiveChannel().
+// or .Unsubscribe()
+type Subscription struct {
+	closed    bool
+	removeSub func()
```

Comment:
Why is removeSub a function pointer? It is always a very specific function. Could it be substituted by PubSubChannel as the field and to have it as the regular method?

Reply:
I guess I just mimic the current implementation that returned a function pointer and when I extended to include "closed" I didn't reconsidered it.

Reply:
Thanks.